payments/schema/mutations.py

```diff
@@ -741,6 +747,118 @@ def mutate_and_get_payload(cls, root, info, order_id, **input):
         return RefundOrderMutation(order_refund=refund)
 
 
+class CreateBerthSwitchOfferMutation(graphene.ClientIDMutation):
+    class Input:
+        # Required fields
+        application_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        # Optional fields
+        old_lease_id = graphene.ID(
+            description="If provided, it will ignore the `Berth Switch` from the application "
+            "and force this lease to be terminated"
+        )
+        due_date = graphene.Date()
+        profile_token = graphene.String(
+            description="API token for Helsinki profile GraphQL API",
+        )
+
+    berth_switch_offer = graphene.Field(BerthSwitchOfferNode, required=True)
+
+    @staticmethod
+    def get_old_lease(application: BerthApplication) -> BerthLease:
+        # TODO: This logic has to be refactored when the old harbors app is removed
+
+        # Based on the information filled by the customer on the switch application,
+        # we retrieve the corresponding lease
+        switch: BerthSwitch = application.berth_switch
+        berth_filters = Q(number__iexact=switch.berth_number)
+
+        # First check if the chosen harbor has a direct map with a resources harbor
+        if harbor := switch.harbor.resources_harbor:
+            berth_filters &= Q(pier__harbor=harbor)
+
+            # Try to match the one with the identifier provided by the customer
+            pier = harbor.piers.filter(identifier__iexact=switch.pier).first()
+
+            # If no pier is found but the harbor only has one pier associated,
+            # we can assume that the old berth has to be on that one pier
+            if not pier and harbor.piers.count() == 1:
+                pier = harbor.piers.first()
+
+            berth_filters &= Q(pier=pier)
+        else:
+            # Fetch all the piers associated to the selected harbor,
+            # in most cases there will be only one pier
+            piers = switch.harbor.resources_pier.all()
+
+            # Try to match the one with the identifier provided by the customer
+            pier = piers.filter(identifier__iexact=switch.pier).first()
+
+            # If no pier is found but the harbor only has one pier associated,
+            # we can assume that the old berth has to be on that one pier
+            if not pier and piers.count() == 1:
+                pier = piers.first()
+
```

Comment:
It will be handled/raised when you try to find a berth that has no pier, so it will raise the `NO_LEASE` error and the frontend will then explicitly pass the lease id>

Reply:
ok